notebooks/theory_model.ipynb

```diff
@@ -0,0 +1,505 @@
+    "        err_list = []\n",
+    "        for fake_mag in fake_mag_list:\n",
+    "            fake_mag_str = str(fake_mag).replace('.', '')\n",
+    "            fake_src = slice_df(fake_src_df, equal={'host_mag': host_mag, 'fake_mag': fake_mag_str})\n",
+    "            eff, err = cal_eff(fake_src)\n",
+    "            efficiency_list.append(eff)\n",
+    "            err_list.append(err)\n",
+    "        eff_dict[host_mag] = efficiency_list\n",
+    "        err_dict[host_mag] = err_list\n",
+    "    return eff_dict, err_dict\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f17a7ca2-b2e6-440d-acf3-cb4ca5fe02b8",
+   "metadata": {},
+   "source": [
+    "## Functions used to calculate the theoretical efficiency"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "6b445001-3488-4fef-81ad-2eb6af5105f6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# estimate the mean variance of an exposure (unit: instFlux^2)\n",
+    "def estimate_var(exp): \n",
+    "    # get the variance plane\n",
+    "    var_arr = exp.getVariance().array\n",
+    "    # remove infinite value and nan value\n",
+    "    inf_idx = np.isinf(var_arr)\n",
+    "    nan_idx = np.isnan(var_arr)\n",
+    "    remove_idx = np.logical_or(inf_idx, nan_idx)\n",
+    "    keep_idx = remove_idx == False\n",
+    "    var_remove = var_arr[keep_idx].copy()\n",
+    "    # remove outliers\n",
+    "    var_clipped, _, _ = sigmaclip(var_remove)\n",
+    "    var = np.median(var_clipped)\n",
+    "    return var\n",
+    "\n",
+    "# convert mags to fluxes with a given photoCalib (unit: number of electrons)\n",
+    "def convert_to_flux(mag_arr, photo_calib):\n",
+    "    gain = 0.7\n",
+    "    arr_length = len(mag_arr)\n",
+    "    flux_list = []\n",
+    "    for mag in mag_arr:\n",
+    "        flux = photo_calib.magnitudeToInstFlux(mag) * gain\n",
+    "        flux_list.append(flux)\n",
+    "    return flux_list\n",
+    "\n",
+    "# calculate the median fluxes (unit: number of electrons)\n",
+    "def cal_median_flux(mag_arr, calexp_butler, calexp_info):\n",
+    "    full_flux_list = []\n",
+    "    for idx, row in calexp_info.iterrows():\n",
+    "        # get calexp id\n",
+    "        visit = int(row['visit'])\n",
+    "        filt = str(row['filter'])\n",
+    "        detector = int(row['detector'])\n",
+    "        data_id = {'visit': visit, 'filter': filt, 'detector': detector}\n",
+    "        # get photoCalib\n",
+    "        photo_calib = calexp_butler.get('calexp_photoCalib',  data_id)\n",
+    "        flux_list = convert_to_flux(mag_arr, photo_calib)\n",
+    "        full_flux_list.append(flux_list)\n",
+    "    full_flux_arr = np.array(full_flux_list)\n",
+    "    median_flux_arr = np.median(full_flux_arr, axis=0)\n",
+    "    return median_flux_arr\n",
+    "\n",
+    "# calculate the median sky variance (unit: number of electrons)\n",
+    "def cal_sky_var(calexp_butler, calexp_info):\n",
```

Comment:
You want the sky variance across all of the pixels in the calibrated aperture.  This is just the sky variance in one pixel.

Reply:
Should I adjust the sky_sacle to the number of pixels in a 4'' aperture and set host_scale to 1?

Reply:
Ah, hmmm... Now that I think about this more what you actually are looking for is the detection, not the final photometry measurement.  So it's not clearly directly related to the photometric calibration aperture.  I realize it was the code that translated a given host galaxy magnitude to counts that led me to think that it was the calibrated aperture, but that's not really it.

What really matters is the counts in the detection.  But doing things in terms of the final photometry calibrated measurement will at least be consistent.  I think the goal here should be to put the two different variance sources on the same level.  And then you can find that "Oh, and there's another factor of ~1.5 that we empirical find that must have to do with exactly how the detection algorithm works."